plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
 	})
 }
 
+func TestValidator_ValidateAllTypedefs(t *testing.T) {
+	v, err := NewValidator(ValidatorOpts{})
+	assert.Nil(t, err)
+	schema, err := ioutil.ReadFile("testdata/all-typedefs.lua")
+	assert.Nil(t, err)
+	pluginName, err := v.LoadSchema(string(schema))
+	assert.Nil(t, err)
+	assert.EqualValues(t, "all-typedefs", pluginName)
+
+	tests := []struct {
+		name        string
+		config      string
+		wantErr     bool
+		expectedErr string
+	}{
+		{
+			name: "valid http method",
+			config: `{
+				"http_method": "GET",
+				"tag": "test"
```

Comment:
Yeah the problem is that the tag field is [required](https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua#L400), so the validation would fail if missing. What we can do is moving the `tag` field to the `methods-and-headers` schema, where we test less fields and so we would have less duplication

Reply:
Or are you suggesting to remove the tag field from the schema entirely?

Reply:
Oh, nice, I didn't know that was possible!

https://github.com/Kong/goks/pull/14/commits/c194fd4a2536a09779ae72f47e4f4ae4452939f4
https://github.com/Kong/goks/pull/14/commits/33162568fad8853f468511ea57e4668e7e3c495d